unit_test/test_spdm_requester/get_digests.c

```diff
@@ -1577,6 +1760,204 @@ static void libspdm_test_requester_get_digests_case26(void **state)
+                            (void *)&spdm_context->connection_info.peer_key_pair_id[index],
+                            sizeof(spdm_key_pair_id_t));
+        assert_memory_equal((void *)&m_local_cert_info[index],
+                            (void *)&spdm_context->connection_info.peer_cert_info[index],
+                            sizeof(spdm_key_pair_id_t));
+        assert_memory_equal((void *)&m_local_key_usage_bit_mask[index],
+                            (void *)&spdm_context->connection_info.peer_key_usage_bit_mask[index],
+                            sizeof(spdm_key_pair_id_t));
+    }
+}
+
+/**
+ * Test 28:
+ * 1.For slot 0, at least one of KeyExUse , ChallengeUse , MeasurementUse , and EndpointInfoUse shall be set. The
+ *   corresponding capability bits shall be set appropriately
+ * 2.In all cases, the certificate model for slot 0 shall be either the device certificate model or the alias certificate model.
+ * Set KeyUsageMask and CertificateInfo to 0
+ * Expected Behavior: requester returns the status LIBSPDM_STATUS_INVALID_MSG_FIELD
+ **/
+static void libspdm_test_requester_get_digests_case28(void **state)
+{
+    libspdm_return_t status;
+    libspdm_test_context_t *spdm_test_context;
+    libspdm_context_t *spdm_context;
+    uint8_t slot_mask;
+    uint8_t total_digest_buffer[LIBSPDM_MAX_HASH_SIZE * SPDM_MAX_SLOT_COUNT];
+
+    spdm_test_context = *state;
+    spdm_context = spdm_test_context->spdm_context;
+    spdm_test_context->case_id = 0x1C;
+    spdm_context->connection_info.version = SPDM_MESSAGE_VERSION_13 <<
+                                            SPDM_VERSION_NUMBER_SHIFT_BIT;
+    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_NEGOTIATED;
+    spdm_context->connection_info.capability.flags |= SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CERT_CAP;
+    spdm_context->connection_info.algorithm.base_hash_algo = m_libspdm_use_hash_algo;
+    libspdm_set_mem(m_libspdm_local_certificate_chain,
+                    sizeof(m_libspdm_local_certificate_chain),
+                    (uint8_t)(0xFF));
+    libspdm_reset_message_b(spdm_context);
+
+#if LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT
+    spdm_context->transcript.message_m.buffer_size =
+        spdm_context->transcript.message_m.max_buffer_size;
+#endif
+
+    spdm_context->connection_info.multi_key_conn_rsp = true;
+    libspdm_reset_message_d(spdm_context);
+
+    /* Sub Case 1: Set KeyUsageMask to 0*/
+    m_local_key_pair_id[0] = 0x00;
+    m_local_cert_info[0] = SPDM_CERTIFICATE_INFO_CERT_MODEL_DEVICE_CERT;
+    m_local_key_usage_bit_mask[0] = 0;
+
+    libspdm_zero_mem(total_digest_buffer, sizeof(total_digest_buffer));
+    status = libspdm_get_digest(spdm_context, NULL, &slot_mask, &total_digest_buffer);
+    assert_int_equal(status, LIBSPDM_STATUS_INVALID_MSG_FIELD);
+
+    /* Sub Case 2: Set CertificateInfo to 0*/
+    m_local_key_pair_id[0] = 0x00;
+    m_local_cert_info[0] = 0;
```

Comment:
0 is valid. If you want to trigger invalid msg, please use GENERIC_CERT.

Reply:
Thanks, I have fix the problem